ideas design + ...sticky sidebar

Roseskk opened the pull request.

Roseskk committed:
ideas design + ...sticky sidebar

Roseskk committed:
ideas design + ...sticky sidebar

Roseskk committed:
need review

BigTows commented on src/components/composed/ideas/constructor/idea_card/IdeaCard.js:
```diff
@@ -0,0 +1,65 @@
+                               <span className={'font-montserratThin text-xs'}>@{idea.ownerIdea.userName}</span>
+                           </div>
+                           <div>
+                               <span className={'font-montserratThin text-xs'}>{new Date(idea.publishDate).toDateString()}</span>
```
Time ago

BigTows commented on src/pages/explore/[page].tsx:
```diff
@@ -22,10 +33,19 @@ export default function Home() {
 export const getServerSideProps: GetServerSideProps = async ({locale}) => {
+    const ideaService = injector.get(IdeaService)
+    const ideas = await ideaService.getIdeasByParams()
+    // const [idea, setIdea] = useState({})
```
Убрать комменты

BigTows commented on src/services/idea/IdeaService.ts:
```diff
@@ -22,6 +22,6 @@ export default class IdeaService {
 
     public getIdeasByParams(): Promise<IdeaSearchResult> {
-        return this.api.getIdeas(1, SortType.Date, ["back-end", "front-end"], ["JavaScript", "Java"])
+        return this.api.getIdeas(1, SortType.Date)
```
Передавать параметры в метод.

Roseskk replied:
это другой тикет где пагинация будет я думаю

Roseskk committed:
need review

Roseskk committed:
need review

Roseskk committed:
need review

BigTows commented on src/components/composed/ideas/constructor/idea_card/IdeaList/Languages.js:
```diff
@@ -1,9 +1,25 @@
+    return(
+        <ul className={'w-36  list-disc flex flex-col  gap-2 mt-1 p-4 rounded bg-white h-24 overflow-y-scroll'}>
+                {
+                    lang.map((requirements,idx)=>{
```
мб как-то проще можно ;/

BigTows replied:
Добавить hashSet на все

BigTows replied:
+ Если пусто, не нужно выводить пункт меню

Roseskk replied:
я хз как проще просто

BigTows commented:
Не хватает пагинации.

Roseskk committed:
need review

Roseskk committed:
need review

BigTows commented on package.json:
```diff
@@ -24,7 +24,8 @@
     "react-redux": "^7.2.0",
-    "redux-thunk": "^2.4.1"
+    "redux-thunk": "^2.4.1",
+    "underscore": "^1.13.2"
```
А тут оно нужно?

Roseskk replied:
не тесты были

BigTows commented on src/services/utility/UtilityService.ts:
```diff
@@ -0,0 +1,15 @@
+    }
+
+    public removeDupedItemObject(arr : Array<any>) : Array<any> {
+        const uniq = new Set(arr.map(e => JSON.stringify(e)));
```
Колхоз какой-то

Roseskk replied:
да ну

BigTows commented on src/services/utility/UtilityService.ts:
```diff
@@ -0,0 +1,15 @@
+
+export default class UtilityService {
+
+    public removeDupedItemArray(arr : Array<any>) : Array<any> {
```
Можно сделать методы статичесикими, потому что у тебя нету состояния, это чисто класс ради алгоритма

BigTows commented on src/config/DependencyInjection.ts:
```diff
@@ -51,6 +52,7 @@ export const injector: DependencyInjector = makeInjector([
     },
     {provide: ProfileService, useClass: ProfileService, deps: [ProfileApi, FileApi, StateManagerService]},
     {provide: IdeaService, useClass: IdeaService, deps: [IdeaApi, FileApi]},
+    {provide: UtilityService, useClass: UtilityService, deps: []},
```
Можно без экземпляра

BigTows commented on src/components/composed/ideas/constructor/idea_card/IdeaList/Specialists.js:
```diff
@@ -1,8 +1,22 @@
-        <div></div>
+        <ul className={'w-36  list-disc flex flex-col  gap-2 mt-1 p-4 rounded bg-white h-24 overflow-y-scroll' }>
+            {
+                service.removeDupedItemObject(specialist).map((spec,idx)=>{
```
https://stackoverflow.com/questions/9229645/remove-duplicate-values-from-js-array
 Может в этом направлении подумать, я просто не помню структуру данных

Roseskk replied:
у нас массив объектов , по этому так сделал

BigTows commented:
И почему-то 500
https://deploy-preview-65--cheerful-lily-b65942.netlify.app/explore/1

Roseskk committed:
need review

Roseskk committed:
need review

Roseskk committed:
need review

Roseskk committed:
complete

Roseskk committed:
complete

BigTows committed:
Refactor logic for Language.
Merge branch 'develop' into 55-подготовить-архитектуру-для-i18n

# Conflicts:
#	src/pages/_app.tsx
#	src/pages/index.js
#	src/pages/users/[userId].tsx

Roseskk committed:
need review ...TS ...refactor ...circle ...time-ago

Roseskk committed:
need review ...TS ...refactor ...circle ...time-ago

BigTows commented on src/components/composed/ideas/constructor/idea_card/IdeaCard.tsx:
```diff
@@ -0,0 +1,70 @@
+                           </div>
+                           <div className={'flex flex-wrap gap-10 mt-4'}>
+                               <div className={'mt-2 flex flex-col font-montserratBold text-x2s text-gray-400'}>
+                                   <span>Специалисты</span>
```
Если нет специалистов, не показывать

BigTows commented on src/components/composed/ideas/constructor/idea_card/IdeaCard.tsx:
```diff
@@ -0,0 +1,70 @@
+                                   <Specialists specialist={idea.requirements} />
+                               </div>
+                               <div className={'mt-2 flex flex-col font-montserratBold text-x2s text-gray-400'}>
+                                   <span>Технологии</span>
```
Если нет технологий, не показывать

BigTows commented on src/components/composed/ideas/constructor/idea_card/IdeaCard.tsx:
```diff
@@ -0,0 +1,70 @@
+                                   <Tech tech={idea.requirements}  />
+                               </div>
+                               <div className={'mt-2 flex flex-col font-montserratBold text-x2s text-gray-400'}>
+                                   <span>Языки программирования</span>
```
Если нет ЯП не показывать

BigTows commented on src/components/composed/ideas/constructor/idea_card/IdeaList/Specialists.tsx:
```diff
@@ -0,0 +1,26 @@
+    return(
+        <ul className={'w-48   flex flex-col  gap-2 mt-1 p-4 rounded bg-white h-24 overflow-y-scroll' }>
+            {
+                UtilityService.removeDupedSpecialistObject(params.specialist).map((spec,idx)=>{
```
map + filter 
Лады, да да

BigTows commented on src/components/composed/ideas/constructor/idea_card/IdeaList/Tech.tsx:
```diff
@@ -0,0 +1,36 @@
+    return(
+        <ul className={'w-48  flex flex-col  gap-2 mt-1 p-4 rounded bg-white h-24 overflow-y-scroll '}>
+            {
+                params.tech.map((requirements)=>{
```
flatmap x2 + map + filter

BigTows commented on src/pages/explore/[page].tsx:
```diff
@@ -21,11 +32,14 @@ export default function Home() {
 
 
-export const getServerSideProps: GetServerSideProps = async ({locale}) => {
+export const getServerSideProps: GetServerSideProps = async (context : any) => {
```
А тут может быть не any?

BigTows commented on src/services/idea/IdeaService.ts:
```diff
@@ -21,7 +21,7 @@ export default class IdeaService {
-    public getIdeasByParams(): Promise<IdeaSearchResult> {
-        return this.api.getIdeas(1, SortType.Date, ["back-end", "front-end"], ["JavaScript", "Java"])
+    public getIdeasByParams(page : number): Promise<IdeaSearchResult> {
+        return this.api.getIdeas(page, SortType.Date,[],[])
```
Добавить и остальные параметры

BigTows commented on src/services/utility/UtilityService.ts:
```diff
@@ -0,0 +1,79 @@
+        return  Array.from(uniq).map(e => JSON.parse(e));
+    }
+
+    static setColorLanguage(language : string | undefined) : string {
```
либо на back либо в random hex

Roseskk replied:
Решим

BigTows commented on src/services/utility/objects/SpecialistsType.ts:
```diff
@@ -0,0 +1,15 @@
+}
+
+export interface Langs {
+    name : string | undefined
```
Удалить - (@Roseskk)

Roseskk committed:
need review ...TS ...refactor ...circle ...time-ago

Roseskk committed:
need review ...TS ...refactor ...circle ...time-ago

Roseskk committed:
need review ...TS ...refactor ...circle ...time-ago

Roseskk committed:
need review ...TS ...refactor ...circle ...time-ago

Roseskk committed:
need review ...TS ...refactor ...circle ...time-ago
